src/devices/LidarLiteV3/LidarLiteV3.cs

```diff
@@ -0,0 +1,319 @@
+
+        /// <summary>
+        /// Get or set the maximum aquisition count limits the number of times
+        /// the device will integrate acquistions to find a correlation
+        /// record peak.
+        /// 
+        /// Roughly correlates to: acq rate = 1/count and max 
+        /// range = count^(1/4)
+        /// </summary>
+        private int MaximumAcquisitionCount {
+            get {
+                Span<byte> rawData = stackalloc byte[1] { 0 };
+                ReadBytes(Register.SIG_COUNT_VAL, rawData);
+                return rawData[0];
+            }
+            set {
+                WriteRegister(Register.SIG_COUNT_VAL, (byte)value);
+            }
+        }
+
+        /// <summary>
+        /// Get or set the threshold of peak value that bypasses the internal algorithm.
+        /// 
+        /// Recommended non-default values are 32 for higher sensitivity
+        /// but higher erronenous measurement and 96 for reduced 
+        /// sensitivity and fewer erroneous measurements.
+        /// </summary>
+        public int AlgorithmBypassThreshold {
+            get {
+                Span<byte> rawData = stackalloc byte[1] { 0 };
+                ReadBytes(Register.THRESHOLD_BYPASS, rawData);
+                return rawData[0];
+            }
+            set {
+                WriteRegister(Register.THRESHOLD_BYPASS, (byte)value);
+            }
+        }
+
+        /// <summary>
+        /// Get or set the power control option.
+        /// </summary>
+        public PowerOptionFlag PowerOption {
+            get {
+                Span<byte> rawData = stackalloc byte[1] { 0 };
+                ReadBytes(Register.POWER_CONTROL, rawData);
+                return (PowerOptionFlag)rawData[0];
+            }
+            set {
+                // Bit 0 disables receiver circuit
+                WriteRegister(Register.POWER_CONTROL, (byte)value);
+            }
+        }
+
+        /// <summary>
+        /// Get the system status
+        /// </summary>
+        public SystemStatusFlag Status {
+            get {
+                Span<byte> rawData = stackalloc byte[1] { 0 };
+                ReadBytes(Register.STATUS, rawData);
+                return (SystemStatusFlag)rawData[0];
+            }
+        }
+
+        #endregion
+
+        #region I2C
+
+        internal void WriteRegister(Register reg, byte data)
+        {
+            Span<byte> dataout = stackalloc byte[] { (byte)reg, data };
+            _i2cDevice.Write(dataout);
+        }
+
+        internal byte ReadByte(Register reg)
+        {
+            _i2cDevice.WriteByte((byte)reg);
+            return _i2cDevice.ReadByte();
+        }
+
+        internal void ReadBytes(Register reg, Span<byte> readBytes)
+        {
+            _i2cDevice.WriteByte((byte)reg);
+            _i2cDevice.Read(readBytes);
+        }
+
+        /// <inheritdoc/>
+        public void Dispose()
+        {
+            if (_gpioController != null && _powerEnablePin.HasValue)
```

Comment:
gpioController will dispose pins in its Dispose method

Reply:
Aren't you suppose to have a single shared instance of `GPIOController` for your application, so it can manage all pin access?

If that's the case, wouldn't you want each disposing devices to release pins it no longer uses, so that other devices in the same application can reuse those pins?  Maybe this is an unrealistic scenario since only one device is physically connected to that pin.

Reply:
ended up removing this, make sense to have controller dipose them.

Reply:
You should create one GpioController per device although I think it doesn't matter that much if you have one or multi because of the static nature of the physical setup